### ISSUE TITLE
Can i show error more detail?

### ISSUE THREAD

**Issue by hhhroot** [dimitri/pgcopydb#640]
I used v0.15.1. And my command is `pgcopydb clone --table-jobs 12 --index-jobs 3 --source {source_db} --target {target_db} --restore-jobs 8 --large-objects-jobs 8 --no-owner --drop-if-exists split-tables-larger-than 250GB --resume --not-consistent`
My process raise below errors.

My database schema is [this](https://blockscout.github.io/blockscout-db-schema/index.html) and It maybe has some extensions and functions.

```
08:31:49.288 60746 INFO   Using work dir "/tmp/pgcopydb"
08:31:49.303 60746 INFO   Exported snapshot "00000008-00001CFA-1" from the source database
08:31:49.304 60748 INFO   STEP 1: fetch source database tables, indexes, and sequences
08:31:51.354 60748 INFO   Fetched information for 51 tables (including 0 tables split in 0 partitions total), with an estimated total of 4855 thousands tuples and 2059 MB on-disk
08:31:52.216 60748 INFO   Fetched information for 174 indexes (supporting 51 constraints)
08:31:52.314 60748 INFO   Fetching information for 28 sequences
08:31:52.445 60748 INFO   Fetched information for 5 extensions
08:31:52.620 60748 INFO   Found 0 indexes (supporting 0 constraints) in the target database
08:31:52.627 60748 INFO   STEP 2: dump the source database schema (pre/post data)
08:31:52.632 60748 INFO    /usr/bin/pg_dump -Fc --snapshot 00000008-00001CFA-1 --section pre-data --file /tmp/pgcopydb/schema/pre.dump 'postgres://postgres@{aws_rds_host}:5432/postgres?keepalives=1&keepalives_idle=10&keepalives_interval=10&keepalives_count=60'
08:31:52.747 60748 INFO    /usr/bin/pg_dump -Fc --snapshot 00000008-00001CFA-1 --section post-data --file /tmp/pgcopydb/schema/post.dump 'postgres://postgres@{aws_rds_host}:5432/postgres?keepalives=1&keepalives_idle=10&keepalives_interval=10&keepalives_count=60'
08:31:52.865 60748 INFO   STEP 3: restore the pre-data section to the target database
08:31:52.909 60748 INFO   Drop tables on the target database, per --drop-if-exists
08:31:52.957 60748 INFO    /usr/bin/pg_restore --dbname 'postgres://xXfswQprtuOqU0G@production-postgres2.cfooe00ckogn.us-east-2.rds.amazonaws.com:5432/okjD3QHPAGS0pTy?keepalives=1&keepalives_idle=10&keepalives_interval=10&keepalives_count=60' --jobs 8 --clean --if-exists --no-owner --use-list /tmp/pgcopydb/schema/pre-filtered.list /tmp/pgcopydb/schema/pre.dump
08:31:53.290 60762 INFO   STEP 4: starting 12 table-data COPY processes
08:31:53.302 60765 INFO   STEP 8: starting 12 VACUUM processes
08:31:53.306 60763 INFO   STEP 6: starting 3 CREATE INDEX processes
08:31:53.306 60763 INFO   STEP 7: constraints are built by the CREATE INDEX processes
08:31:53.315 60748 INFO   Skipping large objects: none found.
08:31:53.329 60748 INFO   STEP 9: reset sequences values
08:31:53.336 60793 INFO   Reset sequences values on the target database
free(): corrupted unsorted chunks
free(): corrupted unsorted chunks
08:31:54.412 60762 ERROR  Sub-process 60764 exited with code 0 and signal Segmentation fault
08:31:55.513 60762 ERROR  Sub-process 60768 exited with code 0 and signal Aborted
08:31:56.514 60762 ERROR  Sub-process 60775 exited with code 0 and signal Aborted
08:31:57.515 60762 ERROR  Sub-process 60767 exited with code 0 and signal Segmentation fault
08:31:58.516 60762 ERROR  Sub-process 60771 exited with code 0 and signal Segmentation fault
malloc(): invalid size (unsorted)
malloc(): invalid size (unsorted)
08:32:00.017 60762 ERROR  Sub-process 60773 exited with code 0 and signal Aborted
08:32:01.118 60762 ERROR  Sub-process 60769 exited with code 0 and signal Aborted
free(): corrupted unsorted chunks
08:32:02.419 60762 ERROR  Sub-process 60770 exited with code 0 and signal Aborted
malloc(): invalid size (unsorted)
08:32:03.921 60762 ERROR  Sub-process 60772 exited with code 0 and signal Aborted
malloc(): invalid size (unsorted)
double free or corruption (out)
08:32:05.122 60762 ERROR  Sub-process 60774 exited with code 0 and signal Aborted
08:32:06.122 60762 ERROR  Sub-process 60776 exited with code 0 and signal Aborted
08:32:06.320 60766 ERROR  [SOURCE 4569] [42601] ERROR:  syntax error at or near ","
08:32:06.320 60766 ERROR  [SOURCE 4569] LINE 1: copy (SELECT id, watchlist_address_id, , , , direction, , ty...
08:32:06.320 60766 ERROR  [SOURCE 4569]                                                ^
08:32:06.320 60766 ERROR  [SOURCE 4569] Context: copy (SELECT id, watchlist_address_id, , , , direction, , type, method, block_number, amount, tx_fee, viewed_at, inserted_at, updated_at, , name, subject, from_address_hash, to_address_hash, transaction_hash FROM ONLY public.account_watchlist_notifications) to stdout
08:32:06.321 60766 ERROR  Failed to copy data for table with oid 29286 and part number 0, see above for details
08:32:06.321 60766 ERROR  BUG: call to pgsql_commit() without holding an open multi statement connection
08:32:06.321 60766 FATAL  Failed to close snapshot "00000008-00001CFA-1" on "(null)"
08:32:07.423 60762 ERROR  Sub-process 60766 exited with code 0 and signal Segmentation fault
08:32:07.423 60762 ERROR  Some COPY worker process(es) have exited with error, see above for details
08:32:07.423 60762 ERROR  Failed to copy table data, see above for details
08:32:07.439 60748 ERROR  Sub-process 60762 exited with code 12
08:32:09.741 60748 ERROR  Some sub-processes have exited with error status, see above for details
08:32:09.741 60748 ERROR  Errors detected, see above for details
08:32:09.741 60748 FATAL  Failed to COPY the data, see above for details
08:32:09.741 60748 ERROR  Failed to clone source database, see above for details
08:32:09.864 60746 ERROR  clone process 60748 has terminated [6]
```

**Comment by dimitri** [comment]
Hi @hhhroot ; please have a look at the documentation online at https://pgcopydb.readthedocs.io/en/latest/ where you will find available options such as `--debug`. Also, that's available through using `pgcopydb clone --help`. More importantly, given the kind of error that you have, please try again with the most recent copy of the source code, I think I just did a couple changes that could be related to the problem at hand here (free/malloc).

**Comment by hhhroot** [comment]
I ran recent version from ghcr with --debug option.
First, there were no malloc related errors - I think it was an arm arch related issue. No error occurred in x86-based ec2. However, again the process fails with no specific logs.

```
03:45:32.695 5149 DEBUG  catalog.c:710             Opening SQLite database "/tmp/pgcopydb/schema/source.db" with lib version 3.34.1
03:45:32.695 5150 SQL    pgsql.c:1513              [TARGET 174] SET statement_timeout TO 0;
03:45:32.695 5149 DEBUG  catalog.c:710             Opening SQLite database "/tmp/pgcopydb/schema/filter.db" with lib version 3.34.1
03:45:32.696 5150 SQL    pgsql.c:1513              [TARGET 174] SET lock_timeout TO 0;
03:45:32.696 5149 DEBUG  catalog.c:710             Opening SQLite database "/tmp/pgcopydb/schema/target.db" with lib version 3.34.1
03:45:32.696 5150 DEBUG  catalog.c:710             Opening SQLite database "/tmp/pgcopydb/schema/source.db" with lib version 3.34.1
03:45:32.696 5150 DEBUG  catalog.c:710             Opening SQLite database "/tmp/pgcopydb/schema/filter.db" with lib version 3.34.1
03:45:32.696 5150 DEBUG  catalog.c:710             Opening SQLite database "/tmp/pgcopydb/schema/target.db" with lib version 3.34.1
...

03:45:32.698 5148 DEBUG  table-data.c:878          copydb_copy_data_by_oid: 28332 public.internal_transactions, part 0
03:45:32.698 5151 DEBUG  table-data.c:878          copydb_copy_data_by_oid: 28257 public.transactions, part 0
03:45:32.699 5150 DEBUG  catalog.c:582             Catalog has been setup for source "postgres://postgres@migrate-test-aurora.cluster-cfooe00ckogn.us-east-2.rds.amazonaws.com:5432/postgres", target "postgres://postgres@172.17.0.1:5432/postgres", snapshot "00000007-0000032B-1"
03:45:32.699 5149 DEBUG  catalog.c:582             Catalog has been setup for source "postgres://postgres@migrate-test-aurora.cluster-cfooe00ckogn.us-east-2.rds.amazonaws.com:5432/postgres", target "postgres://postgres@172.17.0.1:5432/postgres", snapshot "00000007-0000032B-1"
free(): corrupted unsorted chunks
03:45:32.700 5150 DEBUG  table-data.c:878          copydb_copy_data_by_oid: 28304 public.logs, part 0
03:45:32.700 5149 DEBUG  table-data.c:878          copydb_copy_data_by_oid: 28440 public.token_transfers, part 0

...

03:45:32.770 5157 SQL    pgsql.c:404               Disconnecting from [target] "postgres://postgres@172.17.0.1:5432/postgres?keepalives=1&keepalives_idle=10&keepalives_interval=10&keepalives_count=60"
03:45:32.954 5145 ERROR  copydb.c:774              Sub-process 5148 exited with code 0 and signal Segmentation fault
03:45:32.954 5145 ERROR  copydb.c:774              Sub-process 5149 exited with code 0 and signal Segmentation fault
03:45:32.954 5145 ERROR  copydb.c:774              Sub-process 5150 exited with code 0 and signal Segmentation fault
03:45:32.954 5145 ERROR  copydb.c:774              Sub-process 5151 exited with code 0 and signal Aborted
03:45:32.954 5145 DEBUG  copydb.c:730              copydb_wait_for_subprocesses: no more children
03:45:32.954 5145 ERROR  table-data.c:343          Some COPY worker process(es) have exited with error, see above for details
03:45:32.954 5145 DEBUG  indexes.c:611             Send STOP message to CREATE INDEX queue 17
...
```

### My command and information is below.
'>': in pgcopydb cotainer
'$': in ubuntu
```sh
# my command - just for test
> pgcopydb clone --source postgresql://postgres:postgres@{aws_rds_host}:5432/postgres --target postgresql://postgres:postgres@172.17.0.1:5432/postgres --no-owner --drop-if-exists --debug --skip-vacuum

# pgcopydb --version
> 03:51:07.224 5159 INFO   Running pgcopydb version 0.15.6.g7633458 from "/usr/local/bin/pgcopydb"
pgcopydb version 0.15.6.g7633458
compiled with PostgreSQL 13.13 (Debian 13.13-0+deb11u1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
compatible with Postgres 11, 12, 13, 14, 15, and 16

# uname -a
$ Linux ip-20-10-11-11 6.2.0-1017-aws #17~22.04.1-Ubuntu SMP Fri Nov 17 21:07:13 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
```

* I installed postgres-15 in ghcr.io/dimitri/pgcopydb container for fix mismatch of pg_dump.

**Comment by dimitri** [comment]
Should be fixed by #647
